File: src/testingil/webinar/tddlegacycode/e3/addfunctionality/s2/test/Calculator.java
package testingil.webinar.tddlegacycode.e3.addfunctionality.s2.test;

public class Calculator {
	String display = "";
	int lastArgument = 0;
	int result = 0;
	Boolean newArgument = false;
	Boolean shouldReset = true;

	OperationType lastOperation;

	public void press(String key) {
		if (key.equals("C")) {
			display = "";
			return;
		}
		switch (key) {
		case "+":
			handleOpKey(OperationType.Plus);
			break;
		case "/":
			handleOpKey(OperationType.Div);
			break;
		case "=":
			int currentArgument = parseKeyNumber();
			if (lastOperation == OperationType.Plus) {
				display = Integer.toString(lastArgument + currentArgument);
			}
			if (lastOperation == OperationType.Div && currentArgument == 0) {
				display = "Division By Zero Error";
			}
			shouldReset = true;
			break;
		default:
			if (shouldReset) {
				display = "";
				shouldReset = false;
			}
			if (newArgument) {
				display = "";
				newArgument = false;
			}
			display += key;
			break;
		}
	}

	private void handleOpKey(OperationType operationType) {
		lastOperation = operationType;
		lastArgument = parseKeyNumber();
		newArgument = true;
	}

	private int parseKeyNumber() {
		if (display.equals(""))
			return 0;
		return Integer.parseInt(display);
	}

	public String getDisplay() {
		if (display.equals(""))
			return "0";
		return display;
	}

}

File: Python/e3.addfunctionality/s1.refactor/s3.opextract/calculator.py
from enum import Enum


class OperationType(Enum):
    Plus = 1
    Div = 2


class Calculator:
    display = ""
    lastArgument = 0
    result = 0
    newArgument = False
    shouldReset = True
    lastOperation = OperationType.Plus

    def press(self, key):
        if key == "C":
            self.display = ""
            return

        match key:
            case "+":
                self.handleOpKey(OperationType.Plus)
            case "/":
                self.handleOpKey(OperationType.Div)
            case "=":
                currentArgument = self.parseKeyNumber()
                if self.lastOperation == OperationType.Plus:
                    self.display = str(self.lastArgument + currentArgument)
                if self.lastOperation == OperationType.Div and currentArgument == 0:
                    self.display = "Division By Zero Error"
                self.shouldReset = True
            case _:
                if self.shouldReset:
                    self.display = ""
                    self.shouldReset = False
                if self.newArgument:
                    self.display = ""
                    self.newArgument = False
                self.display += key

    def get_display(self):
        if self.display == "":
            return "0"
        return self.display

    def parseKeyNumber(self):
        if self.display == "":
            return 0
        return int(self.display)

    def handleOpKey(self, operationType):
        self.lastOperation = operationType
        self.lastArgument = self.parseKeyNumber()
        self.newArgument = True
